# Joint configuration for humanoid robot
# Motor index mapping from joint names to motor IDs
motor_index:
  j_pan: 19
  j_pelvis_l: 8
  j_pelvis_r: 7
  j_shoulder_l: 2
  j_shoulder_r: 1
  j_tilt: 20
  j_thigh1_l: 10
  j_thigh1_r: 9
  j_high_arm_l: 4
  j_high_arm_r: 3
  j_thigh2_l: 12
  j_thigh2_r: 11
  j_low_arm_l: 6
  j_low_arm_r: 5
  j_tibia_l: 14
  j_tibia_r: 13
  j_gripper_l: 22  # not real
  j_gripper_r: 21  # not real
  j_ankle1_l: 16
  j_ankle1_r: 15
  j_ankle2_l: 18
  j_ankle2_r: 17

# Isaac Sim joint order
sim_joint_names:
  - j_pelvis_l
  - j_pelvis_r
  - j_shoulder_l
  - j_shoulder_r
  - j_thigh1_l
  - j_thigh1_r
  - j_high_arm_l
  - j_high_arm_r
  - j_thigh2_l
  - j_thigh2_r
  - j_low_arm_l
  - j_low_arm_r
  - j_tibia_l
  - j_tibia_r
  - j_ankle1_l
  - j_ankle1_r
  - j_ankle2_l
  - j_ankle2_r
  - j_pan
  - j_tilt
  - j_gripper_r  # not real
  - j_gripper_l  # not real

# Default joint positions (radians)
default_joint_pos:
  j_pan: 0.0
  j_tilt: 0.0
  j_pelvis_l: 0.0
  j_thigh1_l: 0.0
  j_thigh2_l: 0.4
  j_tibia_l: -0.6
  j_ankle1_l: -0.2
  j_ankle2_l: 0.0
  j_pelvis_r: 0.0
  j_thigh1_r: 0.0
  j_thigh2_r: -0.4
  j_tibia_r: 0.6
  j_ankle1_r: 0.2
  j_ankle2_r: 0.0
  j_shoulder_l: 0.0
  j_high_arm_l: 0.7
  j_low_arm_l: -1.0
  j_gripper_l: -0.99
  j_shoulder_r: 0.0
  j_high_arm_r: 0.7
  j_low_arm_r: 1.0
  j_gripper_r: -0.99

# Joint position limits (min, max) in radians
joint_pos_clip:
  j_pelvis_l: [-0.7900, 2.6200]
  j_pelvis_r: [-2.6200, 0.7900]
  j_shoulder_l: [-4.3600, 4.3600]
  j_shoulder_r: [-4.3600, 4.3600]
  j_thigh1_l: [-1.0500, 0.0000]
  j_thigh1_r: [0.0000, 1.0500]
  j_high_arm_l: [-1.7500, 1.5800]
  j_high_arm_r: [-1.7500, 1.5800]
  j_thigh2_l: [-0.5200, 1.7500]
  j_thigh2_r: [-1.7500, 0.5200]
  j_low_arm_l: [-3.1400, -0.0870]
  j_low_arm_r: [0.0870, 3.1400]
  j_tibia_l: [-1.7500, 0.0000]
  j_tibia_r: [0.0000, 1.7500]
  j_ankle1_l: [-0.5200, 0.5200]
  j_ankle1_r: [-0.5200, 0.5200]
  j_ankle2_l: [-0.3500, 0.3500]
  j_ankle2_r: [-0.3500, 0.3500]
  j_pan: [-1.5700, 1.5700]
  j_tilt: [-0.5200, 0.5200]
  j_gripper_l: [-1.5700, 1.5700]
  j_gripper_r: [-1.5700, 1.5700]

trajectory_behaviors:
  1: "hello"
  2: "crouch"
  3: "confident"
  4: "greet"
  5: "foot"
  6: "step_back"
  7: "shake_body"
        